test/cmdlineargs.jl

```diff
@@ -186,14 +205,30 @@ let exename = `$(Base.julia_cmd()) --sysimage-native-code=yes --startup-file=no`
     @test !success(`$exename --history-file=false`)
 
     # --code-coverage
-    @test readchomp(`$exename -E "Bool(Base.JLOptions().code_coverage)"`) == "false"
-    @test readchomp(`$exename -E "Bool(Base.JLOptions().code_coverage)"
-        --code-coverage=none`) == "false"
-
-    @test readchomp(`$exename -E "Bool(Base.JLOptions().code_coverage)"
-        --code-coverage`) == "true"
-    @test readchomp(`$exename -E "Bool(Base.JLOptions().code_coverage)"
-        --code-coverage=user`) == "true"
+    mktempdir() do dir
+        covfile = replace(joinpath(dir, "coverage.info"), "%" => "%%")
+        @test !isfile(covfile)
+        defaultcov = readchomp(`$exename -E "Bool(Base.JLOptions().code_coverage)"`)
+        opts = Base.JLOptions()
+        coverage_file = (opts.output_code_coverage != C_NULL) ?  unsafe_string(opts.output_code_coverage) : ""
+        @test !isfile(covfile)
+        @test defaultcov == string(opts.code_coverage != 0 && (isempty(coverage_file) || occursin("%p", coverage_file)))
+        @test readchomp(`$exename -E "Base.JLOptions().code_coverage"
+            --code-coverage=$covfile --code-coverage=none`) == "0"
+        @test !isfile(covfile)
+        @test readchomp(`$exename -E "Base.JLOptions().code_coverage"
+            --code-coverage=$covfile --code-coverage`) == "1"
+        @test isfile(covfile)
+        rm(covfile)
+        @test readchomp(`$exename -E "Base.JLOptions().code_coverage"
+            --code-coverage=$covfile --code-coverage=user`) == "1"
+        @test isfile(covfile)
+        rm(covfile)
+        @test readchomp(`$exename -E "Base.JLOptions().code_coverage"
+            --code-coverage=$covfile --code-coverage=all`) == "2"
+        @test isfile(covfile)
+        rm(covfile)
+    end
```

Comment:
It would be very cool if we could test parsing of the resultant file to ensure that we don't accidentally break coverage in the future.

Reply:
Yeah, I've checked that `genhtml` is OK with it, but I'm not sure what more to add to verify coverage. I think getting Coverage.jl reports (on PRs, or shortly after) will be a better way to verify the info isn't broken.

Reply:
Could we run coverage mode on some small test file, something like:

```
if get(ENV, "COVERAGATE_TEST") == "true"
    foo()
else
    bar()
end
```

And just make sure that `foo()` and `bar()` are selected at the proper times?  Just to catch egregious errors.

Reply:
Ah, yeah, that's probably clever